packages/terra-action-footer/src/_ActionFooterContainer.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './ActionFooterContainer.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  children: [],
```

Comment:
Because react handles `children` as an opaque data type, i think we'd be better not to have this defined in `defaultProps`

Reply:
Seems like there's inconsistency on this so far..
Setting children defaultProp to `[]`:
![screen shot 2018-03-09 at 2 11 54 pm](https://user-images.githubusercontent.com/34115417/37227720-e133ee46-23a3-11e8-9e67-d237196017f7.png)

Setting children defaultProp to `null`:
![screen shot 2018-03-09 at 2 12 46 pm](https://user-images.githubusercontent.com/34115417/37227793-101b6770-23a4-11e8-9215-3a21ebb7ed12.png)

Setting children defaultProp to `undefined`:
![screen shot 2018-03-09 at 2 14 00 pm](https://user-images.githubusercontent.com/34115417/37227833-2ab1db0a-23a4-11e8-9c1f-aca00bf75ada.png)

Reply:
I think of the above options, `undefined` would be the most appropriate, as I don't think the result from the warroom doc i linked you has had enough time to trickle out into all the components yet, though i know that's the intention going forward. Honestly, i really wouldn't be setting a `defaultProp` on `children`, personally. Because regardless, unless you start doing something with `React.Children.Only()` or `React.Children.Count()` to control the number of incoming children, you truly have no way of _knowing_ what the type `children` is going to be coming in. Hence why it's proptype always is `node`, the most generic (to my knowledge) react type. 

You have 2 situations i see as problematic.
1. You set `defaultProps = { children: [] }`, and say give a single child, then the **actual** type of `children` doesn't align with the default prop value you're supplying, therefore inconsistency.
2. You provide `defaultProps = { children: null }`, nothing **technically** blows up here, but why initialize grouping type items (objects and arrays) to `null`, when you can assign it to an empty version of either of the aforementioned items. 

If you really **want** to supply a `defaultProp` for `children`, i think `undefined` is the most logical. But again, i'd just recommend removing it as that's what it's going to be, if not supplied anyways. And if it is supplied, it's an opaque data type.

Reply:
Yeah, I had that kind of thought process when making the component.. went to look for a precedent and there wasn't a consistent one.

Reply:
Let's set this default-prop value for children to undefined.